### ISSUE TITLE
[Enhancement] Allow for per-filetype config (or buffer-scoped character variables)

### ISSUE THREAD

**Issue by duckpuppy** [dhruvasagar/vim-table-mode#83]
It would be incredibly nice if I could configure table mode to use markdown compatible tables if the filetype is markdown, but use the default style (which I think looks a little nicer) in other circumstances more easily.  If the various character config variables were buffer-scoped rather than strictly global then the user could configure that with autocommands or ftplugin/ files.

For Markdown in particular, maybe that could be baked into the plugin.

**Comment by dhruvasagar** [comment]
Yes that is a good idea. Let me work on it.

**Comment by polyzen** [comment]
FWIW, here's what I use:

```
if (&filetype ==# 'rst')
  let g:table_mode_corner_corner='+'
  let g:table_mode_header_fillchar='='
endif
```

**Comment by dhruvasagar** [comment]
Right, although better is to place that configuration within a ftplugin.

**Comment by polyzen** [comment]
@dhruvasagar, noted.
